src/base/memory_monitor_sr.hpp

```diff
@@ -50,16 +52,25 @@ namespace cubmem
 
     public:
       size_t get_allocated_size (const char *ptr);
+      void add_stat (char *ptr, const size_t size, const char *file, const int line);
+      void sub_stat (char *ptr);
 
     private:
       static std::string make_tag_name (const char *file, const int line);
 
     private:
       std::string m_server_name;
+      mutable std::mutex m_tag_map_mutex;
+      std::unordered_map <std::string, int> m_tag_map;              // tag name <-> tag id
+      std::unordered_map <int, std::atomic <uint64_t>> m_stat_map;  // tag id <-> memory usage
+      std::atomic <uint64_t> m_total_mem_usage;
+      int m_meta_alloc_count;                                       // for checking occupancy of memory used by metainfo space
```

Comment:
You should initialize the added member variables in ctor.

Reply:
Oh I missed it. Done.